encoder/src/main/scala/org/clulab/scala_transformers/encoder/Encoder.scala

```diff
@@ -19,6 +20,19 @@ class Encoder(val encoderEnvironment: OrtEnvironment, val encoderSession: OrtSes
 
     val result: OrtSession.Result = encoderSession.run(inputs)
     val outputs = Math.fromResult(result)
+
+    if(nonLin.isDefined) {
+      for (matrix <- outputs) {
+        for (i <- 0 until Math.rows(matrix)) {
+          val row = Math.row(matrix, i)
+          for (j <- 0 until Math.cols(matrix)) {
+            val orig = Math.get(row, j)
+            Math.set(row, j, nonLin.get.compute(orig))
+          }
+        }
+      }
+    }
```

Comment:
It looks like we don't have a map method.  It might be nice to hide these details and maybe make it more efficient with code like

```scala
  def map(matrix: MathRowMatrix, f: MathValue => MathValue): Unit = {
    val iterator = matrix.iterator(true, 0, 0, matrix.getNumRows, matrix.getNumCols)
    
    while (iterator.hasNext) {
      val oldValue = iterator.next()
      val newValue = f(oldValue)

      iterator.set(newValue)
    }
  }
```
Then the Encoder would have
```scala
    nonLinOpt.foreach { nonLin =>
      outputs.foreach { matrix =>
        Math.map(matrix, nonLin.compute)
      }
    }
```
There will be an example PR with other details soon.